build/lib/src/builder/build_step_impl.dart

```diff
@@ -80,7 +80,8 @@ class BuildStepImpl implements BuildStep {
   @override
   Future writeAsBytes(AssetId id, FutureOr<List<int>> bytes) {
     _checkOutput(id);
-    var done = _futureOrWrite(bytes, (b) => _writer.writeAsBytes(id, b));
+    var done =
+        _futureOrWrite<List<int>>(bytes, (b) => _writer.writeAsBytes(id, b));
```

Comment:
sent that too soon...

The function literal gets in the way.

It would work to have

```dart
Future _someThing(List<int> arg) async {}
...
var done = _futureOrWrite(bytes, _someThing);
```

But once you introduce that function literal the inference gets confused.

Would also work to do

```dart
var done = _futureOrWrite(bytes, (List<int> b) => _writer.writeAsBytes(id, b));
```

Would you prefer that?